### DISCUSSION TITLE
Blink eats up iPad's storage

### DISCUSSION THREAD

**Discussion by kalmarz** [blinksh/blink#1768] [Support] [answered]
### Checklist

- [x] Using latest version on the [App Store](http://itunes.apple.com/app/id1156707581)
- [x] Read the [docs](http://blink.sh/docs/)
- [x] Searched for [existing GitHub issues](https://github.com/blinksh/blink/issues)

### Configuration

Blink v16.1.7.748
iPadOS 16.4.1

### Describe the bug

I've been copying large files (basically videos) with Blink between a remote host and the iPad's local storage (VLC app). What I see happening is Blink slowly eating up the storage. Under Settings -> General -> iPad Sorage I see Blink's data is getting more bloated by each file copied, and the incremental increase equals to the size of the file has just been copied. Besides this, VLC's data is also getting bigger but that's the expected behaviour. And this goes up to reaching the storage limit of the iPad. Moreover, this data gets also backed up with iCloud Backup.

I would expect some kind of garbage collection going on in the backgroud but no, the bloat stays forever. I couldn't find a way of manually removing these files. I looked into Blink's own sandbox but the 2 folders beneath the softlinks (Documents and iCloud) are empty.

Any advice would be greatly appreciated.

**Comment by carloscabanero** [comment, marked as answer]
Hi! Unfortunately, this is a known issue in the old File Provider API, https://github.com/blinksh/blink/discussions/1601#discussioncomment-3954673 & https://github.com/blinksh/blink/issues/1619. 

Some work-arounds until we implement the new version of the API (please leave a thumbs up here as well, so we know how many people is interested https://github.com/blinksh/blink/issues/1717):
- Instead of using the File Provider, for big files it may be better to just use good old scp or sftp. It is also a whole lot faster and you can control the progress.
- You can remove the cache by deleting the provider (slide left in the Files.app section in the Host config) and adding it again.
- The cache is stored in an upper directory, File Provider Storage. You can remove the cache manually, doing `rm -rf ../File\ Provider\ Storage`

**Reply by carloscabanero** [reply]
From 16.1.9, we are now also trying to skip the File Provider Storage from the iCloud backup.

I created https://github.com/blinksh/blink/issues/1789 to track the specifics now during testing.
